src/TextField/index.js

```diff
@@ -148,133 +154,118 @@ type TextFieldPropsT = {
   withTrailingIcon?: React.Node,
   /** By default, props spread to the input. These props are for the component's root container. */
   rootProps?: Object
-} & TextFieldRootPropsT &
-  SimpleTagPropsT;
+} & SimpleTagPropsT;
 
-export const TextField = withMDC({
-  mdcConstructor: MDCTextField,
-  mdcElementRef: true,
-  mdcApiRef: true,
-  defaultProps: {
-    inputRef: noop,
-    disabled: undefined,
-    required: undefined,
-    invalid: undefined,
-    dense: undefined,
-    box: undefined,
-    outlined: undefined,
-    fullwidth: undefined,
-    label: undefined,
-    textarea: undefined
-  },
-  didUpdate: (props, nextProps, api, inst) => {
-    if (
-      props &&
-      (props.textarea !== nextProps.textarea ||
-        props.outlined !== nextProps.outlined ||
-        props.dense !== nextProps.dense)
-    ) {
-      inst.mdcComponentReinit();
-    }
-    if (props && props.invalid !== nextProps.invalid) {
-      api.foundation_.setValid(!nextProps.invalid);
-    }
-  },
-  onMount: (props, api, inst) => {
-    api.foundation_.setValid(!props.invalid);
+export class TextField extends withFoundation({
+  constructor: MDCTextField,
+  adpater: {
+    addClass: addClass(),
+    removeClass: removeClass(),
```

Comment:
Can you think of any component that does not have `addClass()` and `removeClass()`? Maybe putting this into the base class would be a good idea?

Reply:
Not every adapter has to use it. I'm only keeping it here so there isn't any magical overriding of the adapter methods, what you see in these files is what you get. Otherwise your source of truth is whats in this file, whatever MDCFoundation would do, and whatever the MDC class is doing. Just seems harder to reason about.